src/vs/workbench/contrib/inlineChat/browser/inlineChatWidget.ts

```diff
@@ -787,10 +791,33 @@ export class InteractiveEditorZoneWidget extends ZoneWidget {
 		super._relayout(this._computeHeightInLines());
 	}
 
-	override show(where: IPosition): void {
-		super.show(where, this._computeHeightInLines());
+	override show(selectionRange: Range): void {
+		super.show(selectionRange.getEndPosition(), this._computeHeightInLines());
 		this.widget.focus();
 		this._ctxVisible.set(true);
+		this._setMargins(selectionRange);
+	}
+
+	private _setMargins(selectionRange: Range): void {
+		const info = this.editor.getLayoutInfo();
+		const startLineNumber = selectionRange.getStartPosition().lineNumber;
+		const endLineNumber = selectionRange.getEndPosition().lineNumber;
+		const viewModel = this.editor._getViewModel();
+		let indentationLineNumber = endLineNumber;
+		let indentationLevel = viewModel?.getLineFirstNonWhitespaceColumn(endLineNumber);
+		for (let lineNumber = endLineNumber; lineNumber >= startLineNumber; lineNumber--) {
+			const currentIndentationLevel = viewModel?.getLineFirstNonWhitespaceColumn(lineNumber);
```

Comment:
nit but you call `getLineFirstNonWhitespaceColumn` two times for `endLineNumber`